product/specs/2022-01-views/04-ui-requirements-for-views.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Product Specs
+description: Views: UI Requirements for Views
+published: true
+date: 2022-01-13T19:49:54Z
+tags: 
+editor: markdown
+dateCreated: 2022-01-13T19:49:54Z
+---
+
+Based on the models defined for [views](/product/specs/2022-01-views/02-modeling-views.md) and their [columns](/product/specs/2022-01-views/03-modeling-view-columns.md), the first version of Views should support the following features.
+
+I've marked the features we should support as "for alpha release", but I've also listed a few potential future features so that we can think about them while designing code architecture and UI/UX.
+
+# Interacting with Views
+This covers functionality for Views that have already been created.
+
+## Query
+- **For alpha release**: Users should be able to see the query associated with a View after it has been loaded.
+- **Potential future features**:
+	- Users can edit the query associated with a View.
+	- Users can create a new View using a query.
+
+## Columns
+- **For alpha release**: Users should be able to see all columns associated with a view. Each column should show:
+	- Data Type (non-editable)
+	- Sources (non-editable, set when adding a new column to a View)
+	- Formula (non-editable, set when adding a new column to a View)
+	- Link (non-editable, set when adding a new column to a View)
+- **Potential future features**:
+	- Show virtual columns involved in view creation (via CTEs, subqueries, etc.)
+	- Allow editing formula through UI
+	- Allow using SQL to edit formula
+
+## Rows
+- **For alpha release**: 
+	- Users should be able to see the rows associated with a view.
+	- If a cell is a direct representation of a record, users should be able to edit that record via that cell. The entire record should be edited through a form, not just the single item.
+		- "Direct representation" means that the record has only one data source and no formula.
+	- If a column is a direct representation of a record, users should be able to add a new record of the same type via the View UI.
+- **Potential future features**:
+	- Users can edit the query associated with a View.
+	- Users can create a new View using a query.
+
+## Filters
+- **For alpha release**:
+	- Users should be able to see what filters are applied to their View.
+	- Users should be able to edit and delete filters applied to their view in the UI, including basic use cases for columns that are not visible in the View.
```

Comment:
What does "including basic use cases for columns that are not visible in the View" mean?

Reply:
Assuming `SELECT ID, Title FROM Movies WHERE Year > 2000;` is the query for the view, here the "filter" would be `Year > 2000`. Users should be able to edit that even though `Year` is not a visible column in the View.